assets/js/components/product-control/index.js

```diff
@@ -1,54 +1,221 @@
+			variationsLoading: false,
 			loading: true,
 		};
 
 		this.debouncedOnSearch = debounce( this.onSearch.bind( this ), 400 );
+		this.debouncedGetVariations = debounce( this.getVariations.bind( this ), 200 );
+		this.renderItem = this.renderItem.bind( this );
+		this.onProductSelect = this.onProductSelect.bind( this );
 	}
 
 	componentDidMount() {
 		const { selected } = this.props;
 
 		getProducts( { selected } )
-			.then( ( list ) => {
-				this.setState( { list, loading: false } );
+			.then( ( products ) => {
+				products = products.map( ( product ) => {
+					const count = product.variations ? product.variations.length : 0;
+					return {
+						...product,
+						parent: 0,
+						count: count,
+					};
+				} );
+				this.setState( { products, loading: false } );
+			} )
+			.catch( () => {
+				this.setState( { products: [], loading: false } );
+			} );
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		if ( prevState.product !== this.state.product ) {
+			this.debouncedGetVariations();
+		}
+	}
+
+	getVariations() {
+		const { product, variationsList } = this.state;
+		if ( ! product ) {
+			return;
+		}
+		if ( ! variationsList[ product ] ) {
+			this.setState( { variationsLoading: true } );
+		}
+
+		apiFetch( {
+			// Use v4 which returns variation names.
+			path: addQueryArgs( `/wc/v3/products/${ product }/variations`, {
+				per_page: -1,
+			} ),
+		} )
+			.then( ( variations ) => {
+				variations = variations.map( ( variation ) => ( { ...variation, parent: product } ) );
+				this.setState( ( prevState ) => ( {
+					variationsList: { ...prevState.variationsList, [ product ]: variations },
+					variationsLoading: false,
+				} ) );
 			} )
 			.catch( () => {
-				this.setState( { list: [], loading: false } );
+				this.setState( { termsLoading: false } );
 			} );
 	}
 
 	onSearch( search ) {
 		const { selected } = this.props;
 		getProducts( { selected, search } )
-			.then( ( list ) => {
-				this.setState( { list, loading: false } );
+			.then( ( products ) => {
+				this.setState( { products, loading: false } );
 			} )
 			.catch( () => {
-				this.setState( { list: [], loading: false } );
+				this.setState( { products: [], loading: false } );
 			} );
 	}
 
+	onProductSelect( item ) {
+		return () => {
+			this.setState( {
+				product: item.id,
+			} );
+		};
+	}
+
+	renderItem( args ) {
+		const { item, search, depth = 0, isSelected, onSelect } = args;
+		const { product, variationsLoading } = this.state;
+		const classes = [
+			'woocommerce-search-product__item',
+			'woocommerce-search-list__item',
+		];
+		classes.push( `depth-${ depth }` );
+		classes.push( 'is-radio-button' );
+
+		if ( search.length ) {
+			classes.push( 'is-searching' );
+		}
+		if ( depth === 0 && item.parent !== 0 ) {
+			classes.push( 'is-skip-level' );
+		}
+		if ( item.count ) {
+			classes.push( 'is-variable' );
+		}
+
+		if ( ! item.breadcrumbs.length ) {
+			return [
+				<MenuItem // eslint-disable-line
```

Comment:
> The attribute aria-expanded is not supported by the role menuitemradio.

Reply:
hm, I guess it worked in VoiceOver so that's fine, but it would be better to use a proper role 🤔 this is ok for now, but it might be good to circle back to this & the attribute selector to create a better interface for this.